### PR TITLE
106442/legal requirements

### PR DESCRIPTION
**Pull request by dneed-nimble**
### Context
Added tests for the LR work.
### Checklist

- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: Data.TRAMS.Tests/Mappers/Request/InternalProjectToUpdateMapperTests.cs [hunk 1]
```diff
@@ -166,6 +175,13 @@ private static void AssertDatesAreCorrect(Project toMap, TramsProjectUpdate resu
             Assert.True(result.Dates.HasHtbDate);
             Assert.True(result.Dates.HasTargetDateForTransfer);
         }
+        private static void AssertLegalRequirementsAreCorrect(Project toMap, TramsProjectUpdate result)
```
Comment (jordan-beard) [inline Data.TRAMS.Tests/Mappers/Request/InternalProjectToUpdateMapperTests.cs:178]: Should be on the next line.

File: Data.TRAMS.Tests/Mappers/Response/TramsProjectMapperTests.cs [hunk 2]
```diff
@@ -140,7 +148,13 @@ private static void AssertTransferringAcademiesCorrect(TramsProject toMap, Proje
             Assert.Equal(expectedTransfer.OutgoingAcademy.Ukprn, result.TransferringAcademies[0].OutgoingAcademyUkprn);
             Assert.Equal(expectedTransfer.OutgoingAcademy.Urn, result.TransferringAcademies[0].OutgoingAcademyUrn);
         }
-
+        private static void AssertLegalRequirementsAreCorrect(TramsProject toMap, Project result)
```
Comment (jordan-beard) [inline Data.TRAMS.Tests/Mappers/Response/TramsProjectMapperTests.cs:151]: Should be on the next line.

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {
+        }
+
+        [Fact]
+        public async Task Should_successfully_load_radio_buttons()
+        {
+            var project = GetProject(p => p.LegalRequirements.DiocesanConsent = ThreeOptions.No.ToDescription());
+
+            await OpenUrlAsync($"/project/{project.ProjectUrn}/legalrequirements/diocesan-consent");
+
```
Comment (jordan-beard) [inline Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs:27]: remove line 27

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+namespace Frontend.Integration.Tests.Pages.Projects.LegalRequirements
+{
+    public class DiocesanConsentTests : BaseIntegrationTests
+    {
+        public DiocesanConsentTests(IntegrationTestingWebApplicationFactory factory) : base(factory)
+        {
+        }
+
+        [Fact]
+        public async Task Should_successfully_load_radio_buttons()
```
Comment (jordan-beard) [inline Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs:22]: Should there be a test for submitting the form?

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/FoundationConsentTests .cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Data.TRAMS.ExtensionMethods;
+using Xunit;
+using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
+
+namespace Frontend.Integration.Tests.Pages.Projects.LegalRequirements
+{
+    public class FoundationConsentTests : BaseIntegrationTests
```
Comment (jordan-beard) [inline Frontend.Integration.Tests/Pages/Projects/LegalRequirements/FoundationConsentTests .cs:15]: Should there be a test for submitting the form?

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        {
+        }
+
+        [Fact]
+        public async Task Should_successfully_load_radio_buttons()
+        {
+            var project = GetProject(p => p.LegalRequirements.TrustAgreement = ThreeOptions.No.ToDescription());
+
+            await OpenUrlAsync($"/project/{project.ProjectUrn}/legalrequirements/trust-agreement");
+
```
Comment (jordan-beard) [inline Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs:27]: remove line 27

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Data.TRAMS.ExtensionMethods;
+using Xunit;
+using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
+
+namespace Frontend.Integration.Tests.Pages.Projects.LegalRequirements
+{
+    public class TrustAgreementIntegrationTests : BaseIntegrationTests
```
Comment (jordan-beard) [inline Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs:15]: Should there be a test for submitting the form?

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+            public async void GivenUrn_FetchesProjectFromTheRepository()
+            {
+                _subject.Urn = ProjectUrn0001;
+                await _subject.OnGetAsync();
+
+                ProjectRepository.Verify(r => r.GetByUrn(ProjectUrn0001), Times.Once);
+            }
+
+            [Fact]
+            public async void GivenReturnToPreview_KeepsValue()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs:39]: Should return an empty Task.

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                await _subject.OnGetAsync();
+
+                Assert.True(_subject.ReturnToPreview);
+            }
+        }
+
+        public class PostTests : DiocesanConsentTests
+        {
+            [Fact]
+            public async void GivenUrnAndDiocesanConsent_UpdatesTheProject()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs:52]: Should return an empty Task.

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+            {
+
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs:54]: remove line 54

File: Data.TRAMS.Tests/Mappers/Response/TramsProjectMapperTests.cs [hunk 11]
```diff
@@ -101,7 +102,13 @@ public void GivenTramsProject_MapsCorrectly()
                         KeyStage5PerformanceAdditionalInformation = "KeyStage5PerformanceAdditionalInformation"
                     }
                 },
-                
+                LegalRequirements = new AcademyTransferProjectLegalRequirements()
+                {
+                    TrustAgreement = "No",
+                    DiocesanConsent = "No",
+                    FoundationConsent = "No",
+                    IsCompleted= false,
```
[the remark this answers is not in the document]
Comment (dneed-nimble) [reply Data.TRAMS.Tests/Mappers/Response/TramsProjectMapperTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+                await _subject.OnPostAsync();
+
+                ProjectRepository.Verify(r =>
+                    r.Update(It.Is<Project>(
+                        project => project.LegalRequirements.DiocesanConsent == ThreeOptions.No)));
+            }
+
+
+            [Fact]
+            public async void GivenUrnAndDiocesanConsent_RedirectsToTheSummaryPage()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs:66]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+                _subject.DiocesanConsentViewModel.DiocesanConsent = ThreeOptions.No;
+
+                var response = await _subject.OnPostAsync();
+
+                ControllerTestHelpers.AssertResultRedirectsToPage(response, "/Projects/LegalRequirements/Index",
+                    new RouteValueDictionary(new { Urn = ProjectUrn0001 }));
+            }
+
+            [Fact]
+            public async void GivenReturnToPreview_RedirectToThePreviewPage()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs:79]: Return empty Task

File: Data.TRAMS.Tests/Mappers/Response/TramsProjectMapperTests.cs (hunk 2 above)
Comment (dneed-nimble) [reply Data.TRAMS.Tests/Mappers/Response/TramsProjectMapperTests.cs:152]: Done

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 3 above)
Comment (dneed-nimble) [reply Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs:27]: Done

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 4 above)
Comment (dneed-nimble) [reply Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs:22]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+            public async void GivenUrn_FetchesProjectFromTheRepository()
+            {
+                _subject.Urn = ProjectUrn0001;
+                await _subject.OnGetAsync();
+
+                ProjectRepository.Verify(r => r.GetByUrn(ProjectUrn0001), Times.Once);
+            }
+
+            [Fact]
+            public async void GivenReturnToPreview_KeepsValue()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs:39]: Return empty Task

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs (hunk 6 above)
Comment (dneed-nimble) [reply Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs:27]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+                await _subject.OnGetAsync();
+
+                Assert.True(_subject.ReturnToPreview);
+            }
+        }
+
+        public class PostTests : FoundationConsentTests
+        {
+            [Fact]
+            public async void GivenUrnAndFoundationConsent_UpdatesTheProject()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs:52]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+            {
+
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs:54]: remove line 54

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+                await _subject.OnPostAsync();
+
+                ProjectRepository.Verify(r =>
+                    r.Update(It.Is<Project>(
+                        project => project.LegalRequirements.FoundationConsent == ThreeOptions.No)));
+            }
+
+
+            [Fact]
+            public async void GivenUrnAndFoundationConsent_RedirectsToTheSummaryPage()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs:66]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+                _subject.FoundationConsentViewModel.FoundationConsent = ThreeOptions.No;
+
+                var response = await _subject.OnPostAsync();
+
+                ControllerTestHelpers.AssertResultRedirectsToPage(response, "/Projects/LegalRequirements/Index",
+                    new RouteValueDictionary(new { Urn = ProjectUrn0001 }));
+            }
+
+            [Fact]
+            public async void GivenReturnToPreview_RedirectToThePreviewPage()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs:79]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+
+        public TrustAgreementTests()
+        {
+            _subject = new TrustAgreementModel(ProjectRepository.Object);
+        }
+
+        public class GetTests : TrustAgreementTests
+        {
+            [Fact]
+            public async void GivenUrn_FetchesProjectFromTheRepository()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs:30]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+                await _subject.OnGetAsync();
+
+                Assert.True(_subject.ReturnToPreview);
+            }
+        }
+
+        public class PostTests : TrustAgreementTests
+        {
+            [Fact]
+            public async void GivenUrnAndTrustAgreement_UpdatesTheProject()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs:52]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+            {
+
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs:54]: Remove line 54

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+                await _subject.OnPostAsync();
+
+                ProjectRepository.Verify(r =>
+                    r.Update(It.Is<Project>(
+                        project => project.LegalRequirements.TrustAgreement == ThreeOptions.No)));
+            }
+
+
+            [Fact]
+            public async void GivenUrnAndTrustAgreement_RedirectsToTheSummaryPage()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs:66]: Return empty Task

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+                _subject.TrustAgreementViewModel.TrustAgreement = ThreeOptions.No;
+
+                var response = await _subject.OnPostAsync();
+
+                ControllerTestHelpers.AssertResultRedirectsToPage(response, "/Projects/LegalRequirements/Index",
+                    new RouteValueDictionary(new { Urn = ProjectUrn0001 }));
+            }
+
+            [Fact]
+            public async void GivenReturnToPreview_RedirectToThePreviewPage()
```
Comment (jordan-beard) [inline Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs:79]: Return empty Task

**Review by jordan-beard — CHANGES_REQUESTED** [review]
Fast work 😀
Added a few comments, though most of them are more stylistic preferences.

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 8 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 9 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 10 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs:54]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 12 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 13 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/DiocesanConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+
+        public FoundationConsentTests()
+        {
+            _subject = new FoundationConsentModel(ProjectRepository.Object);
+        }
+
+        public class GetTests : FoundationConsentTests
+        {
+            [Fact]
+            public async void GivenUrn_FetchesProjectFromTheRepository()
```
[the remark this answers is not in the document]
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs (hunk 14 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs (hunk 15 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs (hunk 16 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs:54]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs (hunk 18 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/FoundationConsentTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs (hunk 19 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+            public async void GivenUrn_FetchesProjectFromTheRepository()
+            {
+                _subject.Urn = ProjectUrn0001;
+                await _subject.OnGetAsync();
+
+                ProjectRepository.Verify(r => r.GetByUrn(ProjectUrn0001), Times.Once);
+            }
+
+            [Fact]
+            public async void GivenReturnToPreview_KeepsValue()
```
[the remark this answers is not in the document]
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs (hunk 21 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs:54]: Done

File: Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs (hunk 22 above)
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/LegalRequirements/TrustAgreementTests.cs]: Done

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs (hunk 4 above)
Comment (dneed-nimble) [reply Frontend.Integration.Tests/Pages/Projects/LegalRequirements/DiocesanConsentTests.cs:24]: Yeah, sounds good. Added.

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/FoundationConsentTests .cs (hunk 5 above)
Comment (dneed-nimble) [reply Frontend.Integration.Tests/Pages/Projects/LegalRequirements/FoundationConsentTests .cs:15]: Yeah, sounds good. Added.

File: Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs (hunk 7 above)
Comment (dneed-nimble) [reply Frontend.Integration.Tests/Pages/Projects/LegalRequirements/TrustAgreementTests.cs:15]: Yeah, sounds good. Added.

File: Frontend.Integration.Tests/Pages/Projects/TaskList/PreviewTests.cs [hunk 26]
```diff
@@ -0,0 +1,39 @@
+﻿using AngleSharp.Dom;
```
Comment (jordan-beard) [inline Frontend.Integration.Tests/Pages/Projects/TaskList/PreviewTests.cs:1]: Lovely test 😁

**Review by jordan-beard — APPROVED** [review]

**Pull request merged**